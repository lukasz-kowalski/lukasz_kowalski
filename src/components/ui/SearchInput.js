import React from 'react';

class SearchInput extends React.Component {
  handleChange = event => {
    this.props.handleQueryChange(event);
  }

  render() {
    const { id, label, placeholder } = this.props;
    return (
      <div className="form-group">
        <label htmlFor={id} className="sr-only">{label}</label>
        <input
          type="search"
          className="form-control" 
          id={id}
          placeholder={placeholder} 
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

export default SearchInput;
