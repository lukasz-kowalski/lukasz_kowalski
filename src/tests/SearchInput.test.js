import React from 'react';
import { shallow } from 'enzyme';
import SearchInput from '../components/ui/SearchInput';
import { wrap } from 'module';

let wrapper;

const handleChange = jest.fn();
const handleQueryChange = jest.fn();

beforeEach(() => {
  wrapper = shallow(
  <SearchInput
    id="searchInput"
    label="Search characters..."
    isValid={true}
    placeholder="Search characters..."
    handleQueryChange={handleQueryChange}
  />);
})

test('should successful render SearchInput', () => {
  expect(wrapper).toMatchSnapshot();
})

test('should call handleQueryChange when calling handleChange', () => {
  const event = { target: { value: 'Test' } };
  wrapper.instance().handleChange(event);
  expect(handleQueryChange).toBeCalledWith(event);
})

test('should call handleQueryChange function on change', () => {
  const event = { target: { value: 'Test' } };
  wrapper.find('#searchInput').simulate('change', event);
  expect(handleQueryChange).toBeCalledWith(event);
})
