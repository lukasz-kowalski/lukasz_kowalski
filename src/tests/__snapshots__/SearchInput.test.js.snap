// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render SearchInput 1`] = `
<div
  className="form-group"
>
  <label
    className="sr-only"
    htmlFor="searchInput"
  >
    Search characters...
  </label>
  <input
    className="form-control"
    id="searchInput"
    onChange={[Function]}
    placeholder="Search characters..."
    type="search"
  />
</div>
`;
